components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -640,6 +649,35 @@ public static String retrieveValidatedGraphqlSchemaFromArchive(String pathToArch
         }
     }
 
+    /**
+     * Retrieve graphql complexity information from the file and validate it with the schema
+     *
+     * @param pathToArchive Path to API archive
+     * @param schema        GraphQL schema
+     * @throws APIImportExportException If an error occurs while reading the file
+     */
+    private static GraphqlComplexityInfo retrieveGraphqlComplexityInfoFromArchive(String pathToArchive, String schema) {
+
+        try {
+            String jsonContent =
+                    getFileContentAsJson(pathToArchive + ImportExportConstants.GRAPHQL_COMPLEXITY_INFO_LOCATION);
+            if (jsonContent == null) {
+                return null;
+            }
+
+            JsonElement configElement = new JsonParser().parse(jsonContent).getAsJsonObject().get(APIConstants.DATA);
+            GraphQLQueryComplexityInfoDTO complexityDTO = new Gson().fromJson(String.valueOf(configElement),
+                    GraphQLQueryComplexityInfoDTO.class);
+            GraphqlComplexityInfo graphqlComplexityInfo =
+                    GraphqlQueryAnalysisMappingUtil.fromDTOtoValidatedGraphqlComplexityInfo(complexityDTO, schema);
+            return graphqlComplexityInfo;
+        } catch (IOException e) {
+            log.error("Error reading the GraphQL complexity information from the file.");
+        }
+        return null;
+    }
+
+
```

Comment:
Shall we remove this newline?

Reply:
Fixed with 2adb0a8ed29a163835d83b3c7f1d3f809b27c2d7